### PR TITLE
Project: Ruby Enumerable Methods

### PR DESCRIPTION
**Pull request by OlukaDenis**

### CODE REVIEW

File: enumerable_methods.rb [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+      result = args[0]
+    else
+      result = args[0]
+      symbol = args[1]
+    end
+    arr[0..-1].my_each do |elem|
+      result = if symbol
+                 result.send(symbol, elem)
+               else
+                 yield(resultrubco, elem)
```
Comment (maelfosso) [inline enumerable_methods.rb]: There is an error here
```ruby
irb(main):034:0> operation = proc { |sum, n| sum + n }
=> #<Proc:0x0000558a4162a630@(irb):34>
irb(main):035:0> array.inject(&operation)
=> 361
irb(main):036:0> array.my_inject(&operation)
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):36
        3: from enumerable_methods.rb:117:in `my_inject'
        2: from enumerable_methods.rb:14:in `my_each'
        1: from enumerable_methods.rb:121:in `block in my_inject'
NameError (undefined local variable or method `resultrubco' for #<Array:0x0000558a41681cf0>)
Did you mean?  result
irb(main):037:0>
```

File: enumerable_methods.rb [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      my_each { |element| return true if element }
+    elsif arg.class == Class
+      my_each { |element| return true if element.class.ancestors.include? arg }
+    elsif arg.class == Regexp
+      my_each { |element| return true if element =~ arg }
+    else
+      my_each { |element| return true if element == arg }
+    end
+    true
+  end
```
Comment (maelfosso) [inline enumerable_methods.rb]:
```ruby
irb(main):006:0> array
=> [2, 8, 5, 5, 1, 5, 2, 4, 5, 0, 1, 2, 6, 3, 6, 3, 6, 5, 1, 2, 0, 3, 0, 6, 2, 2, 7, 0, 2, 1, 5, 3, 3, 0, 5, 1, 2, 5, 2, 2, 1, 2, 2, 0, 1, 2, 3, 6, 7, 4, 4, 3, 0, 8, 1, 2, 7, 2, 7, 0, 4, 7, 5, 2, 7, 1, 5, 8, 0, 7, 8, 3, 8, 5, 8, 1, 1, 4, 7, 2, 0, 3, 7, 5, 7, 3, 3, 3, 8, 1, 2, 0, 3, 3, 8, 3, 5, 5, 8, 6]
irb(main):007:0> 
irb(main):008:0> words = %w[dog door rod blade]
=> ["dog", "door", "rod", "blade"]
irb(main):009:0> 
```

Comment (maelfosso) [reply enumerable_methods.rb]:
```ruby
irb(main):010:0> false_block = proc { |num| num > 9 }
=> #<Proc:0x0000558a4163c448@(irb):10>
irb(main):011:0> array.any?(&false_block)
=> false
irb(main):012:0> array.my_any?(&false_block)
=> true
irb(main):013:0>
```

Comment (maelfosso) [reply enumerable_methods.rb]:
```ruby
irb(main):015:0> false_array = [nil, false, nil, false]
=> [nil, false, nil, false]
irb(main):016:0> false_array.any?
=> false
irb(main):017:0> false_array.my_any?
=> true
irb(main):018:0>
```

Comment (maelfosso) [reply enumerable_methods.rb]:
```ruby
irb(main):019:0> words
=> ["dog", "door", "rod", "blade"]
irb(main):020:0> words.any?(Integer)
=> false
irb(main):021:0> words.my_any?(Integer)
=> true
irb(main):022:0>
```

Comment (maelfosso) [reply enumerable_methods.rb]:
```ruby
irb(main):024:0> words.any?(/z/)
=> false
irb(main):025:0> words.my_any?(/z/)
=> true
irb(main):026:0> 
```

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required

Hello @OlukaDenis 

Nice job :smile: Awesome code :clap: 
However there are some bugs you should fix. 

Also, **Follow the Git Flow** otherwise the project won't be approved.
- Do not work directly in the master branch. Create the development branch and a specific feature-branch (give it a descriptive name) for every feature you implement. When your feature is ready and approved you can merge feature-branch to the development branch and when the release is ready create a pull request from development to master.
- Read more [here](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59882084/)

Please, read the comments below and fix them then submit a new code review request.
Thanks

**Mael FOSSO**
[Github](https://github.com) | [Twitter](https://twitter.com) | [LinkedIn](https://www.linkedin.com/in/mael-elvis-fosso-650b6346/)

File: enumerable_methods.rb (hunk 2 above)
Comment (maelfosso) [reply enumerable_methods.rb]:
```ruby
irb(main):047:0> words
=> ["cat", "door", "rod", "blade"]
irb(main):048:0> words.any?('dog')
=> false
irb(main):049:0> words.any?('dog')
=> false
irb(main):050:0> words.my_any?('dog')
=> true
irb(main):051:0>
```

**Comment by OlukaDenis** [comment]
About the git flow, have i made a mistake anywhere :grin: . Please look at this attachment
git flow

File: enumerable_methods.rb (hunk 1 above)
Comment (OlukaDenis) [reply enumerable_methods.rb]: Error resolved

File: enumerable_methods.rb (hunk 2 above)
Comment (OlukaDenis) [reply enumerable_methods.rb]: Error resolved

**Comment by OlukaDenis** [comment]
Hello there, I have corrected the errors. Would you please checkout

File: enumerable_methods.rb [hunk 3]
```diff
@@ -54,16 +54,16 @@ def my_all?(arg = nil)
   def my_any?(arg = nil)
     if block_given?
       my_each { |element| return true if yield(element) }
-    elsif arg.nil?
-      my_each { |element| return true if element }
     elsif arg.class == Class
-      my_each { |element| return true if element.class.ancestors.include? arg }
+      my_each { |element| return true if element.class == arg }
     elsif arg.class == Regexp
```
Comment (samgaco) [inline enumerable_methods.rb:59]: :medal_military:

**Review by samgaco — APPROVED** [review]
Everything is perfect now :+1:  You can move on.

------------------------
* ###### STATUS OF THE PROJECT: APPROVED :heavy_check_mark:

**Pull request merged**